domain/robotouille.json

```diff
@@ -1004,6 +1278,142 @@
             "precons": [],
             "immediate_fx": [],
             "sfx": []
+        },
+        {
+            "name": "add-ketchup",
+            "precons": [
+                {
+                    "predicate": "has_condiment",
+                    "params": ["p1", "cd1"],
+                    "is_true": true
+                },
+                {
+                    "predicate": "isketchup",
+                    "params": ["cd1"],
+                    "is_true": true
+                },
+                {
+                    "predicate": "isgrilled",
```

Comment:
i see what you did here for the hotdog task but this would limit ketchup to being added to items that can only be grilled (what about adding ketchup or mustard to a cooked patty). however, you also don't want ketchup or mustard to be applied on a raw hotdog sausage. @lsuyean has dealt with this in the past for any fried goods by creating a predicate, `fryableifcut`. discuss with him about the possibility of creating a `condimentifgrilled` predicate

Reply:
added new predicates!